tensorpack/dataflow/image.py

```diff
@@ -17,6 +17,17 @@ def _valid_coords(coords):
     assert coords.shape[1] == 2, coords.shape
     assert np.issubdtype(coords.dtype, np.float), coords.dtype
 
+class ExceptionHandler:
+    def __init__(self, allow_exceptions=False):
+        self._nr_error = 0
+        self.allow_exceptions = allow_exceptions
+
+    def catch(self):
```

Comment:
You can make it a contextmanager, so the `try ... catch` logic can be shared as well. 
An example: https://github.com/ppwwyyxx/tensorpack/blob/2fa49895399ec33298f2b774929aada5fbed3e89/tensorpack/dataflow/prefetch.py#L51-L65.